File: 알고리즘 공부/프로그래머스 js 연습/옹알이.py
babbling = ["ayaye", "uuuma", "ye", "yemawoo", "ayaa"];
babbling1 = ["aya", "yee", "u", "maa", "wyeoo"];
baby = ["aya", "ye", "woo", "ma"];

cnt = 0
arr = []
arr1 = []

for i in babbling:
  for j in baby:  # baby와 babbling 비교
    if j in i: # 연속적인 배열 없는 경우
      i=i.replace(j, '1') # babbling[i]의 baby값이 있으면 순서대로 '1'로 변환
  print([i])
  if i.replace('1',''): # 문자 '1' 공백처리
    arr1.append(i)
    continue
  else: #1로 이루어진 경우 cnt값 증가
    cnt += 1
    arr.append(i)

print(arr1)
print(arr)
print(cnt)
print("hello world")

File: 알고리즘 공부/프로그래머스 js 연습/옹알이.js
const babbling = ["ayaye", "uuuma", "ye", "yemawoo", "ayaa"];
const babbling1 = ["aya", "yee", "u", "maa", "wyeoo"];
const baby = ["aya", "ye", "woo", "ma"];

let count = 0;
const arr = [];
const arr1 = [];
for (let i of babbling1) {
  for (let j of baby) {
    i = i.replace(j, "#", 1);
    arr.push(i);
    // i = i.replace(j, "!");
    if (arr.includes(j)) {
      count += 1;
    }
  }
}
console.log(arr);
console.log(babbling1);
console.log(baby);
console.log(count);
